### PR TITLE
Replace ConstantLike with ConstantOfShape in symbolic

### PR DESCRIPTION
**Pull request by zrphercule**
We recently replaced ConstantLike with ConstantOfShape in ONNX, 
https://github.com/onnx/onnx/pull/1716
https://github.com/onnx/onnx/pull/1582
therefore we should stop using this op in pytorch exporter as well. 
Notice since we do not have importer of ConstantLike related ops in caffe2, and this op is not included in our upcoming ONNX release, we dont have any in/external production breaking.

For now, since the ONNX version in thirdparty is not new enough to have ConstantOfShape, this diff is NOT READY to be merged (until we updated ONNX in thirdparty). But review is always welcomed =)

### CODE REVIEW

**Comment by zrphercule** [comment]
The onnx of pytorch has been updated, which means it is ok to merge this pr now (if it passed CI)

File: torch/onnx/symbolic.py [hunk 1]
```diff
@@ -1060,6 +1060,16 @@ def _cast_func_template(to_i, g, input, non_blocking):
     name = '_cast_{}'.format(k)
     globals()[name] = parse_args('v', 'i')(partial(_cast_func_template, v))
 
+pytorch_type_to_scalar_type = [
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: I think hardcode here is fine, but please add some comments to explain why it's fine :-)

File: torch/onnx/symbolic.py [hunk 2]
```diff
@@ -1081,7 +1091,10 @@ def zeros(g, sizes, dtype, layout, device):
 
 @parse_args('v', 'i', 'v', 'v')
 def zeros_like(g, input, dtype, layout, device):
-    return g.op("ConstantLike", input, dtype_i=scalar_type_to_onnx[dtype], value_f=0.0)
+    shape = g.op("Shape", input)
+    print(dtype, torch.float32)
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: oops, print?

**Review by houseroad — COMMENTED** [review]
Clean the code a bit?

**Comment by spandantiwari** [comment]
@zrphercule  - good to see that onnx submodule has been updated. Thanks. 
There may be some other sites where `ConstantFill` is being used. One usage is in the optimization pass `torch\csrc\jit\passes\onnx\peephole.cpp` in `fixDefaultRNNState()` method. Should we update that in this PR or a separate one?

File: torch/onnx/symbolic.py [hunk 3]
```diff
@@ -1107,7 +1122,9 @@ def full(g, sizes, value, dtype, layout, device):
 
 @parse_args('v', 'f', 'i', 'v', 'v')
 def full_like(g, input, fill_value, dtype, layout, device):
-    return g.op("ConstantLike", input, dtype_i=scalar_type_to_onnx[dtype], value_f=fill_value)
+    shape = g.op("Shape", input)
+    return g.op("ConstantOfShape", shape,
```
Comment (spandantiwari) [inline torch/onnx/symbolic.py]: Good to see this change.

**Comment by houseroad** [comment]
@spandantiwari unfortunately, @zrphercule is off until next week, feel free to take over this PR. I will be the last backup.

**Pull request closed without merging**